server/services/curiousService.ts

```diff
@@ -30,11 +36,44 @@ export default class CuriousService {
       const learnerProfiles = await this.getLearnerProfile(prisonNumber, systemToken)
       return toFunctionalSkills(learnerProfiles)
     } catch (error) {
-      logger.error(`Error retrieving data from Curious: ${error}`)
+      logger.error(`Error retrieving data from Curious: ${JSON.stringify(error)}`)
       return { problemRetrievingData: true } as FunctionalSkills
     }
   }
 
+  /**
+   * Returns the specified prisoner's Education Records
+   *
+   * The Curious `learnerEducation` API is a paged API. This function calls the API starting from page 0 until there are no
+   * more pages remaining. The cumulative array of Curious `LearnerEducation` records from all API calls are mapped into
+   * and an array of `PrisonerEducation` within the returned `PrisonerEducationRecords` object.
+   */
+  async getLearnerEducation(prisonNumber: string, username: string): Promise<PrisonerEducationRecords> {
+    const systemToken = await this.hmppsAuthClient.getSystemClientToken(username)
+
+    try {
+      let page = 0
+      let apiPagedResponse = { last: false } as LearnerEductionPagedResponse
+      const apiLearnerEducation: Array<LearnerEducation> = []
+
+      // loop until the API response's `last` field is `true`
+      while (apiPagedResponse.last === false) {
+        // eslint-disable-next-line no-await-in-loop
```

Comment:
`eslint` frowns upon using an `await` inside a loop because it says it is inefficient.
It's rationale is that you instead you return all your promises in a `Promise.all` ... which does indeed make sense .... if your loop is not controlled/dependent on the result of each promise.

Our loop _is_ controlled by the result of each promise - we need the result of the promise (ie. the API call to Curious to get the page of data) in order to decide whether its the last page or not, and whether to keep looping.

It basically makes our API call synchronous rather than async, which is the main objection to using `await` in a loop.
But because we need to know the result of the previous API call, we are implicitly synchronous 🤷‍♂️